rskj-core/src/test/java/co/rsk/peg/federation/PendingFederationTest.java

```diff
@@ -138,95 +140,257 @@ void testToString() {
+    void buildFederation_with6Members_afterRSKIP201Activation_beforeRSKIP284Activation_inTestnet_shouldBuildNonStandardERPFed() {
+        // Arrange
+        long federationCreationBlockNumber = 0L;
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.iris300().forBlock(federationCreationBlockNumber);
+        FederationArgs federationArgs = new FederationArgs(federationMembers, creationTime, federationCreationBlockNumber, btcParamsFromTestnetConstants);
+        List<BtcECKey> erpPubKeys = federationTestnetConstants.getErpFedPubKeysList();
+        long activationDelay = federationTestnetConstants.getErpFedActivationDelay();
+
+        Federation expectedFederation = FederationFactory.buildNonStandardErpFederation(federationArgs, erpPubKeys, activationDelay, activations);
+
+        // Act
+        PendingFederation otherPendingFederation = new PendingFederation(
+            federationMembers
+        );
+        Federation builtFederation = otherPendingFederation.buildFederation(
+            creationTime,
+            federationCreationBlockNumber,
+            federationTestnetConstants,
+            activations
         );
+
+        // Assert
+        assertEquals(expectedFederation, builtFederation);
+        assertEquals(NON_STANDARD_ERP_REDEEM_SCRIPT_HARDCODED, builtFederation.getRedeemScript());
     }
 
     @Test
-    void buildFederation_erp_ok_after_RSKIP_201_activation_before_RSKIP284_mainnet() {
-        testBuildFederation(
-            true,
-            false,
-            false,
-            NetworkParameters.ID_MAINNET,
-            6
+    void buildFederation_with6Members_afterRSKIP201Activation_beforeRSKIP284Activation_inMainnet_shouldBuildNonStandardERPFed() {
+        // Arrange
+        long federationCreationBlockNumber = 0L;
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.iris300().forBlock(federationCreationBlockNumber);
+        FederationArgs federationArgs = new FederationArgs(federationMembers, creationTime, federationCreationBlockNumber, btcParamsFromMainnetConstants);
+        List<BtcECKey> erpPubKeys = federationMainnetConstants.getErpFedPubKeysList();
+        long activationDelay = federationMainnetConstants.getErpFedActivationDelay();
+        Federation expectedFederation = FederationFactory.buildNonStandardErpFederation(federationArgs, erpPubKeys, activationDelay, activations);
+
+        // Act
+        PendingFederation otherPendingFederation = new PendingFederation(
+            federationMembers
         );
+        Federation builtFederation = otherPendingFederation.buildFederation(
+            creationTime,
+            federationCreationBlockNumber,
+            federationMainnetConstants,
+            activations
+        );
+
+
+        // Assert
+        assertEquals(expectedFederation, builtFederation);
+    }
+
+    @Test
+    void buildFederation_with6Members_afterRSKIP201AndRSKIP284Activations_inTestnet_shouldBuildNonStandardERPFed() {
```

Comment:
```suggestion
    void buildFederation_with6Members_afterRSKIP284Activations_inTestnet_shouldBuildNonStandardERPFed() {
```

probably no need to say its after rskip201

Reply:
done

Reply:
done